### ISSUE TITLE
Limits in manifest file size

### ISSUE THREAD

**Issue by tafkam** [argoproj/argo-cd#3998]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

App status is "Unknown" and refreshing all the time.

**To Reproduce**

Create ArgoCD app with cert-manager 0.16.0 manifests and it's absurdly big CRD manifests (~1mb) 
https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml

**Expected behavior**

kubectl CLI can handle the manifests (1.17.9), so should ArgoCD. When the CRD ressources are removed from the yaml file and Argocd pods are restarted, the app behaves as intended.

**Screenshots**

**Version**

```shell
# argocd version
argocd: v1.6.1+159674e
  BuildDate: 2020-06-19T00:39:46Z
  GitCommit: 159674ee844a378fb98fe297006bf7b83a6e32d2
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
argocd-server: v1.6.1+159674e
  BuildDate: 2020-06-19T00:41:05Z
  GitCommit: 159674ee844a378fb98fe297006bf7b83a6e32d2
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: {Version:kustomize/v3.6.1 GitCommit:c97fa946d576eb6ed559f17f2ac43b3b5a8d5dbd BuildDate:2020-05-27T20:47:35Z GoOs:linux GoArch:amd64}
  Helm Version: version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
  Kubectl Version: v1.14.0

```

**Logs**

```shell
argocd app get cert-manager
Name:               cert-manager
Project:            cluster
Server:             https://kubernetes.default.svc
Namespace:          cert-manager
URL:                https://1.1.1.1/applications/cert-manager
Repo:               git@xxxxxxxx.xxx/xxx.git
Target:             HEAD
Path:               cluster/app-resources/cert-manager
SyncWindow:         Sync Allowed
Sync Policy:        <none>
Sync Status:
Health Status:
```

reposerver log (no more references to cert-manager but this):
```
time="2020-07-25T17:10:23Z" level=info msg="kustomize build /tmp/git@xxxxxxxx.xxx/xxx/cluster/app-resources/cert-manager" dir= execID=CSJzX
```

**Comment by jessesuen** [comment]
Need to check if still valid after updating our k8s libraries to v1.17

**Comment by MrFreezeex** [comment]
I have a similar issue running the latest commit at the time I am posting.
I am running kubernetes 1.18 so It could get even worse with managedfields.

Cert-manager crds are successfully installed but argocd still shows CRDs as "Unknown" in the UI and If I check on "diff" It shows me the whole CRD and If i click on the "live manifest" tabs argocd also shows me the whole CRD (meaning It can get It but can't generate a proper diff ?).

If I restart argocd I have the same issue, the whole application is marked as "Uknown".

What is really strange is I have a similar issue with kubectl: I can get my CRDs, I can print a whole CRD with -o yaml but a kubectl diff of some cert-manager CRDs (like clusterissuers.cert-manager.io) does not works.

I did have a working deployment of cert-manager a few days ago but this issue arose when I update my cert-manager manifests recently.

**Comment by tafkam** [comment]
this issue was probably fixed with 1.7.2 #4105
